pyk/src/tests/integration/kllvm/test_prooftrace.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+from typing import cast
+
+import pyk.kllvm.load  # noqa: F401
+import pyk.kllvm.prooftrace as prooftrace
+from pyk.testing import ProofTraceTest
+
+
+class TestProofTrace(ProofTraceTest):
+    KOMPILE_DEFINITION = """
+        module TEST-PROOF-TRACE-SYNTAX
+          syntax Foo ::= a() | b() | c()
+        endmodule
+        
+        module TEST-PROOF-TRACE
+          imports TEST-PROOF-TRACE-SYNTAX
+          rule a() => b()
+          rule b() => c()
+        endmodule
+    """
+    KOMPILE_MAIN_MODULE = 'TEST-PROOF-TRACE'
+    HINTS_INPUT_KORE = r"""
+        LblinitGeneratedTopCell{}(
+          Lbl'Unds'Map'Unds'{}(
+            Lbl'Stop'Map{}(),
+            Lbl'UndsPipe'-'-GT-Unds'{}(
+               inj{SortKConfigVar{}, SortKItem{}}(\dv{SortKConfigVar{}}("$PGM")),
+               inj{SortFoo{}, SortKItem{}}(Lbla'LParRParUnds'TEST-PROOF-TRACE-SYNTAX'Unds'Foo{}())
+            )
+          )
+        )
+    """
+
+    def test_proof_trace(self, hints: bytes) -> None:
+        pt = prooftrace.LLVMRewriteTrace.parse(hints)
+        assert pt is not None
+
+        # check that there is a initial configuration
+        assert pt.initial_config.is_kore_pattern()
+
+        # check that the trace after the initial configuration is 3 events long
+        assert len(pt.trace) == 3
+
+        # check that the first event is the rewrite a() => b()
+        assert pt.trace[0].is_step_event()
+        rewrite_event = cast('prooftrace.LLVMRewriteEvent', pt.trace[0].step_event)
+        assert rewrite_event.rule_ordinal == 96
```

Comment:
Do you know how stable these rule ordinals are? I'm worried this assertion might break if we make a change to `domains.md`.

Reply:
I'm not sure about changes in `domain.md`, but I would say that they can be a little bit unstable regarding the content of the definition. For instance, I had to modify several assertions like that on Pi2 regression tests when Theo modified the hints not to print intermediate steps! I believe we're using this style here to mimic what we use there. I am unsure if it is ideal from the K/Pyk side, but for us, it could be helpful to identify earlier that these will be affected, so we have to prepare ourselves to fix it when updating the K version.

Reply:
> I am unsure if it is ideal from the K/Pyk side

I would prefer not to introduce K tests that we know are likely to fail when unrelated code is changed; we've been squashing bugs like this recently.

> it could be helpful to identify earlier that these will be affected

Hard-coding the ordinals like this is convenient for writing test cases and has worked thus far, but perhaps now is the right time to figure out a more durable solution. Perhaps there's a way to use / improve the `compute-ordinal` script from the LLVM backend so that ordinals can be computed as part of the test script? Happy to chat through what this might look like if I'm being unclear.